lib/SIL/Verifier/LinearLifetimeChecker.cpp

```diff
@@ -168,58 +177,39 @@ struct State {
 
 void State::initializeAllNonConsumingUses(
     ArrayRef<Operand *> nonConsumingUsers) {
+  SWIFT_DEFER {
+    // Once we have finished initializing blocksWithNonConsumingUses, we need to
+    // freeze it. By using a defer here, we can make sure we don't forget to do
+    // this below.
+    blocksWithNonConsumingUses.setFrozen();
+  };
+
   for (Operand *use : nonConsumingUsers) {
     auto *userBlock = use->getUser()->getParent();
 
-    // First check if this non consuming user is in our definition block. If so,
```

Comment:
In Word: the big change here is instead of trying to find the last element, we insert all non-consuming uses not in the def block into the non consuming use array. We will then identify any issues via the data flow. If a non-consuming use is in the def block though, we make sure it is /after/ the def. If so, we just add it to the multi-map. Otherwise, we emit a special use before def error.